tests/crypto/test_cases/test_ecjpake.c

```diff
@@ -0,0 +1,359 @@
+	msg_srv_1_len = hex2bin(p_test_vector->p_round_message_server_1, strlen(p_test_vector->p_round_message_server_1), m_msg_srv_1, strlen(p_test_vector->p_round_message_server_1));
+	msg_srv_2_len = hex2bin(p_test_vector->p_round_message_server_2, strlen(p_test_vector->p_round_message_server_2), m_msg_srv_2, strlen(p_test_vector->p_round_message_server_2));
+}
+
+void ecjpake_given_setup(void)
+{
+	static int i;
+
+	clear_buffers();
+	p_test_vector = ITEM_GET(test_vector_ecjpake_given_data, test_vector_ecjpake_t, i++);
+	unhexify_ecjpake();
+}
+
+void ecjpake_random_setup(void)
+{
+	static int i;
+
+	clear_buffers();
+	p_test_vector = ITEM_GET(test_vector_ecjpake_random_data, test_vector_ecjpake_t, i++);
+	unhexify_ecjpake();
+}
+
+void ecjpake_ctx_init(mbedtls_ecjpake_context *ctx, mbedtls_ecjpake_role role)
+{
+	int err_code;
+
+	mbedtls_ecjpake_init(ctx);
+	err_code = mbedtls_ecjpake_setup(ctx,
+									 role,
+									 MBEDTLS_MD_SHA256,
+									 MBEDTLS_ECP_DP_SECP256R1,
+									 m_password,
+									 password_len);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	err_code = mbedtls_ecjpake_check(ctx);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+}
+
+void exec_test_case_ecjpake_given(void)
+{
+	int err_code;
+	size_t len_ss_cli;
+	size_t len_ss_srv;
+
+	mbedtls_ecjpake_context ctx_client;
+	mbedtls_ecjpake_context ctx_server;
+
+	ecjpake_ctx_init(&ctx_client, MBEDTLS_ECJPAKE_CLIENT);
+	ecjpake_ctx_init(&ctx_server, MBEDTLS_ECJPAKE_SERVER);
+
+	err_code = ecjpake_test_load(&ctx_client,
+								 m_priv_key_cli_1,
+								 priv_key_cli_1_len,
+								 m_priv_key_cli_2,
+								 priv_key_cli_2_len);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	err_code = ecjpake_test_load(&ctx_server,
+								 m_priv_key_srv_1,
+								 priv_key_srv_1_len,
+								 m_priv_key_srv_2,
+								 priv_key_srv_2_len);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	/* Round one. */
+	err_code = mbedtls_ecjpake_read_round_one(&ctx_server,
+											  m_msg_cli_1,
+											  msg_cli_1_len);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	err_code = mbedtls_ecjpake_read_round_one(&ctx_client,
+											  m_msg_srv_1,
+											  msg_srv_1_len);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	/*
+	The keys are already given; go straight to reading next round.
+	*/
+
+	/* Round two. */
+	err_code = mbedtls_ecjpake_read_round_two(&ctx_client,
+											  m_msg_srv_2,
+											  msg_srv_2_len);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	err_code = mbedtls_ecjpake_read_round_two(&ctx_server,
+											  m_msg_cli_2,
+											  msg_cli_2_len);
+	TEST_VECTOR_ASSERT_EQUAL(0, err_code);
+
+	/* Derive secrets. */
+	err_code = mbedtls_ecjpake_derive_secret(&ctx_client,
+											 m_secret_cli,
+											 sizeof(m_secret_cli),
+											 &len_ss_cli,
+											 mbedtls_ctr_drbg_random,
+											 &ctr_drbg_ctx);
```

Comment:
Can you run a code formatter on the entire PR?

Reply:
Yes, is there a go-to formatter in use?
Could not see a reference to one in the upstream contributor guidelines at least.

Reply:
I believe most use checkpatch and some use clang-format.

Reply:
Formatting applied in e735f14 , using clang-format and style via `fw-nrfconnect-nrf/.clang-format`.
Some alignment breaks as a result of this which might affect readability.
Personally I prefer no changes when running clang-format over having "proper" alignment, so I am fine with keeping this commit as-is.